http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterpSpec.scala

```diff
@@ -359,4 +361,262 @@ class RuntimeServiceInterpSpec extends FlatSpec with LeonardoTestSuite with Test
 
     res.unsafeRunSync()
   }
+
+  it should "update autopause" in isolatedDbTest {
+    val userInfo = UserInfo(OAuth2BearerToken(""), WorkbenchUserId("userId"), WorkbenchEmail("user1@example.com"), 0) // this email is white listed
+
+    val res = for {
+      internalId <- IO(RuntimeInternalId(UUID.randomUUID.toString))
+      testRuntime <- IO(makeCluster(1).copy(internalId = internalId).save())
+      req = UpdateRuntimeRequest(None, false, Some(true), Some(120.minutes))
+
+      _ <- runtimeService.updateRuntime(userInfo, testRuntime.googleProject, testRuntime.runtimeName, req)
+      dbRuntimeOpt <- clusterQuery
+        .getActiveClusterByNameMinimal(testRuntime.googleProject, testRuntime.runtimeName)
+        .transaction
+      dbRuntime = dbRuntimeOpt.get
+      messageOpt <- publisherQueue.tryDequeue1
+    } yield {
+      dbRuntime.autopauseThreshold shouldBe 120
+      messageOpt shouldBe None
+    }
+
+    res.unsafeRunSync()
+  }
+
+  "RuntimeServiceInterp.processUpdateRuntimeConfigRequest" should "not update a GCE runtime when there are no changes" in {
+    val req = UpdateRuntimeConfigRequest.GceConfig(Some(gceRuntimeConfig.machineType), Some(gceRuntimeConfig.diskSize))
+    val res = for {
+      traceId <- traceId.ask
+      _ <- runtimeService.processUpdateRuntimeConfigRequest(req, false, testCluster, gceRuntimeConfig, traceId)
+      messageOpt <- publisherQueue.tryDequeue1
+    } yield {
+      messageOpt shouldBe None
+    }
+    res.unsafeRunSync()
+  }
+
+  it should "update a GCE machine type in Stopped state" in {
+    val req = UpdateRuntimeConfigRequest.GceConfig(Some(MachineTypeName("n1-micro-2")), None)
+    val res = for {
+      traceId <- traceId.ask
+      _ <- runtimeService.processUpdateRuntimeConfigRequest(req,
+                                                            false,
+                                                            testCluster.copy(status = RuntimeStatus.Stopped),
+                                                            gceRuntimeConfig,
+                                                            traceId)
+      message <- publisherQueue.dequeue1
+    } yield {
+      message shouldBe UpdateRuntimeMessage(testCluster.id,
+                                            Some(MachineTypeName("n1-micro-2")),
+                                            false,
+                                            None,
+                                            None,
+                                            None,
+                                            Some(traceId))
+    }
+    res.unsafeRunSync()
+  }
+
+  it should "update a GCE machine type in Running state" in {
```

Comment:
Would it be good to test states other than `Running` and `Stopped`?

Reply:
Like Starting, Stopping, Creating, Deleting? Hm maybe we should disallow updates in those states (and test for it).

Reply:
makes sense to me